Tests/TuistGeneratorTests/Graph/GraphTests.swift

```diff
@@ -588,6 +634,56 @@ final class GraphTests: TuistUnitTestCase {
         XCTAssertEqual(got, expected)
     }
 
+    func test_embeddableDependencies_whenHostedTestTarget() throws {
+        // Given
+        let framework = Target.test(name: "Framework",
+                                    product: .framework)
+
+        let app = Target.test(name: "App", product: .app)
+        let tests = Target.test(name: "AppTests", product: .unitTests)
+        let projectA = Project.test(path: "/path/a")
+
+        let graph = Graph.create(project: projectA,
+                                 dependencies: [
+                                     (target: app, dependencies: [framework]),
+                                     (target: framework, dependencies: []),
+                                     (target: tests, dependencies: [app]),
+                                 ])
+
+        // When
+        let result = try graph.embeddableFrameworks(path: projectA.path, name: tests.name)
+
+        // Then
+        XCTAssertTrue(result.isEmpty)
+    }
+
+    func test_embeddableDependencies_whenHostedTestTarget_transitiveDepndencies() throws {
+        // Given
+        let framework = Target.test(name: "Framework",
+                                    product: .framework)
+
+        let staticFramework = Target.test(name: "StaticFramework",
+                                          product: .framework)
+
+        let app = Target.test(name: "App", product: .app)
+        let tests = Target.test(name: "AppTests", product: .unitTests)
+        let projectA = Project.test(path: "/path/a")
+
+        let graph = Graph.create(project: projectA,
+                                 dependencies: [
+                                     (target: app, dependencies: [staticFramework]),
```

Comment:
Might be worth testing the case below too if it already isn't present:
```                            
dependencies: [
            (target: app, dependencies: [framework]),
            (target: tests, dependencies: [app, framework]),
])
```

Reply:
Good idea, just pushed an update which includes it